addon/-private/column-tree.js

```diff
@@ -0,0 +1,783 @@
+        return set(this, 'column.width', oldWidth + delta);
+      } else {
+        return set(this, '_tempWidth', oldWidth + delta);
+      }
+    } else {
+      let subcolumns = get(this, 'subcolumnNodes');
+
+      // Delta can only be rendered at a pixel level of precision in tables in
+      // some browsers, so we round and distribute the remainder as well. We also
+      // don't know when we may hit a constraint (e.g. minWidth) so we have to do
+      // this repeatedly. We take the largest chunk we can and try to fit it into
+      // each piece in a loop.
+
+      // We distribute chunks to the columns starting from the column with the
+      // smallest width to the column with the largest width.
+      let sortedSubcolumns = subcolumns.sortBy('width').reverse();
+
+      while (delta !== 0) {
+        let deltaChunk = delta / get(subcolumns, 'length');
+        deltaChunk = delta > 0 ? Math.floor(deltaChunk) : Math.ceil(deltaChunk);
+
+        if (deltaChunk === 0) {
+          deltaChunk = delta > 0 ? 1 : -1;
+        }
+
+        for (let subcolumn of sortedSubcolumns) {
+          let oldWidth = get(subcolumn, 'width');
+          let targetWidth = oldWidth + deltaChunk;
+
+          set(subcolumn, 'width', targetWidth);
+
+          let newWidth = get(subcolumn, 'width');
+
+          // subtract the amount that changed, if any
+          delta -= newWidth - oldWidth;
+
+          if (delta === 0) {
+            break;
+          }
+        }
+      }
+
+      return get(this, 'width');
+    }
+  }
+
+  @computed('parent.{offsetLeft,subcolumnNodes.@each.width}')
+  get offsetLeft() {
+    let parent = get(this, 'parent');
+
+    if (!parent) {
+      return 0;
+    }
+
+    let subcolumns = get(parent, 'subcolumnNodes');
+    let offsetLeft = get(parent, 'offsetLeft');
+
+    for (let column of subcolumns) {
+      if (column === this) {
+        break;
+      }
+
+      offsetLeft += get(column, 'width');
+    }
+
+    return offsetLeft;
+  }
+
+  @computed('parent.{offsetRight,subcolumnNodes.@each.width}')
+  get offsetRight() {
+    let parent = get(this, 'parent');
+
+    if (!parent) {
+      return 0;
+    }
+
+    let subcolumns = get(parent, 'subcolumnNodes')
+      .slice()
+      .reverse();
+    let offsetRight = get(parent, 'offsetRight');
+
+    for (let column of subcolumns) {
+      if (column === this) {
+        break;
+      }
+
+      offsetRight += get(column, 'width');
+    }
+
+    return offsetRight;
+  }
+
+  registerElement(element) {
+    this.element = element;
+  }
+}
+
+export default class ColumnTree extends EmberObject {
+  constructor() {
+    super(...arguments);
+
+    this.token = new Token();
+
+    scheduler.schedule('sync', this.ensureWidthConstraint, this.token);
+
+    this.sortColumnsByFixed();
+    addObserver(this, 'columns.@each.isFixed', this.sortColumnsByFixed);
+    addObserver(this, 'widthConstraint', this.ensureWidthConstraint);
+  }
+
+  destroy() {
```

Comment:
On this `EmberObject` should you also be calling `this._super()` in `destroy`?

Reply:
probably should, good catch!